metagraph/src/cli/transform_annotation.cpp

```diff
@@ -427,23 +427,33 @@ int transform_annotation(Config *config) {
             exit(1);
         }
 
+        const bool sum_counts = config->count_kmers;
         const bool filter_values = config->min_value > 1
                                     || config->max_value
                                         < std::numeric_limits<unsigned int>::max();
-        const uint64_t min_cols
-            = std::max((uint64_t)std::ceil(num_columns * config->min_fraction),
-                       (uint64_t)config->min_count);
-        const uint64_t max_cols
-            = std::min((uint64_t)std::floor(num_columns * config->max_fraction),
-                       (uint64_t)config->max_count);
-
-        // TODO: set width to log2(max_cols + 1) but make sure atomic
-        //       increments don't overflow
-        sdsl::int_vector<> sum(0, 0, sdsl::bits::hi(num_columns) + 1);
+        
+        uint64_t min_threshold = config->min_count, max_threshold = config->max_count;
+        if (sum_counts) {
+            if (config->min_fraction > 0.0 || config->max_fraction < 1.0) {
+                logger->warn("--min-fraction and --max-fraction are ignored when --count-kmers is used. "
+                            "Use --min-count and --max-count to filter by aggregated k-mer counts.");
```

Comment:
Better throw an error and exit. There is no need to ever save runs with badly passed parameters. Better force the user to correct the parameters. Otherwise they might have wrong expectations and miss or neglect the warning at all.
```suggestion
                logger->error("--min-fraction and --max-fraction are ignored when --count-kmers is used. "
                            "Use --min-count and --max-count to filter by aggregated k-mer counts.");
                exit(1);
```